### PR TITLE
hypervisor: rework VP state components

### PR DESCRIPTION
**Pull request by russell-islam**
On Microsoft Hypervisor, we need to save/restore five VP state components which are as follows:
	1. Local APIC
	2. Xsave
	3. Synthetic Message Page
	4. Synthetic Event Flags Page
	5. Synthetic Timers

In the MSHV crate we created a single struct for all the components and API to get/set the states.

### CODE REVIEW

**Comment by russell-islam** [comment]
I have tested this PR using supported Dom0 Kernel. Ran Snapshot/restore integration test. It looks fine.

**Comment by russell-islam** [comment]
@NunoDasNeves Could you please review at your convenience?

File: hypervisor/src/mshv/x86_64/mod.rs [hunk 1]
```diff
@@ -38,10 +38,9 @@ pub struct VcpuMshvState {
     pub sregs: MshvSpecialRegisters,
     pub fpu: FpuState,
     pub xcrs: ExtendedControlRegisters,
-    pub lapic: LapicState,
```
Comment (jinankjain) [inline hypervisor/src/mshv/x86_64/mod.rs:41]: We should also clean up the code on line 290 which helps in converting LapicState and MshvLapicState.

Comment (russell-islam) [reply hypervisor/src/mshv/x86_64/mod.rs:41]: LocalApic is required to stay as is. That's because LAPIC is needed separately during VM configuration.

**Review by NunoDasNeves — APPROVED** [review]
LGTM

**Review by jinankjain — APPROVED** [review]

**Comment by russell-islam** [comment]
@likebreath Could you please take a look?

File: hypervisor/src/cpu.rs [hunk 2]
```diff
@@ -115,6 +115,16 @@ pub enum HypervisorCpuError {
...
     ///
+    /// Getting the VP state components error
+    ///
+    #[error("Failed to get VP State Components: {0}")]
+    GetAllVpStateComponents(#[source] anyhow::Error),
+    ///
+    /// Setting the VP state components error
+    ///
+    #[error("Failed to set VP State Components: {0}")]
+    SetAllVpStateComponents(#[source] anyhow::Error),
```
Comment (likebreath) [inline hypervisor/src/cpu.rs:126]: Can you put them behind the `mshv` feature gate given they are mshv specific?

Comment (likebreath) [reply hypervisor/src/cpu.rs:126]: Of course, feel free to put `Set/GetXsaveState` error behind `kvm` feature gate if they are not going to be used by `mshv` by design.

Comment (russell-islam) [reply hypervisor/src/cpu.rs]: Thank you. Addressed your comments.

**Pull request merged**